swagger2.0/oic.r.rule.action.swagger.json

```diff
@@ -0,0 +1,191 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "OCF Rule Action",
+    "version": "20190910",
+    "license": {
+      "name": "OCF Data Model License",
+      "url": "https://github.com/openconnectivityfoundation/core/blob/e28a9e0a92e17042ba3e83661e4c0fbce8bdc4ba/LICENSE.md",
+      "x-copyright": "copyright 2019 Open Connectivity Foundation, Inc. All rights reserved."
+    },
+    "termsOfService": "https://openconnectivityfoundation.github.io/core/DISCLAIMER.md"
+  },
+  "schemes": ["http"],
+  "consumes": ["application/json"],
+  "produces": ["application/json"],
+  "paths": {
+    "/RuleActionResURI": {
+      "get": {
+        "description": "Rule Action contains a link to Scene Collection and a value to be set for the \"lastScene\" Property within that Collection.",
+        "parameters": [
+          {"$ref": "#/parameters/interface-rw"}
+        ],
+        "responses": {
+          "200": {
+            "description" : "Retrieves the Rule action.",
+            "x-example":
+            {
+              "rt": ["oic.r.rule.action"],
+              "if": ["oic.if.rw", "oic.if.baseline"],
+              "link": [{"href": "/myscene","rt":["oic.wk.scenecollection"],"if":["oic.if.a","oic.if.ll","oic.if.baseline"]}],
+              "lastscene": "desiredlastscenevalue"
+            },
+            "schema": { "$ref": "#/definitions/ruleaction" }
+          }
+        }
+      },
+      "post": {
+        "description": "",
+        "parameters": [
+          {"$ref": "#/parameters/interface-rw"},
+          {
+            "name": "body",
+            "in": "body",
+            "required": true,
+            "schema": { "$ref": "#/definitions/ruleaction" },
+            "x-example":
+              {
+                "lastscene": "somedifferentlastscenevalue"
+              }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description" : "",
+            "x-example": {
+              "link": [{"href": "/myscene","rt":["oic.wk.scenecollection"],"if":["oic.if.a","oic.if.ll","oic.if.baseline"]}],
```

Comment:
In this example, you changed the "scenevalue".  If you wanted to UPDATE the "link", I guess you'd have to include the "href", "rt" and "if" at a minimum, right?  I assume that none of the other Link Parameters need to be included, do they?  In other words, what value would it be to include "p" or "rel" as part of this "link" Property.  
It seems like this should be spelled out in the CR about which sub-Properties are required in the "link" Property.

Reply:
We're not updating the link.  We're updating what the link is pointing to.  The link is to a locally hosted scene.  It's a standard OCF Link, I'm not sure why you think otherwise.

Reply:
My problem is that oic.r.rule.action exposes oic.if.rw and we don't have provisions for using that OCF Interface to add or modify Links.  So if "link" is a standard OCF Link, then I think we need this Resource to support oic.if.linkadd and oic.if.linkremove to add and remove Links to the "link" Property.

If we expect Clients to use "oic.if.rw" to manipulate the "link" Property, then we need to define how that is supposed to work either in the Rules CR or in the Core Spec.